src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```
@@ -277,34 +433,55 @@ public void hangup(String conferenceId) throws Exception {
-                if (conferenceIsActive != null && conferenceIsActive) {
-                    return generateCallStatus(getJsonProperty(response, "id", String.class), CallStatus.CallStatusState.Connected);
-                }
-            } else {
-                ArrayNode conferenceCalls = listConferenceCalls();
-                if(conferenceCalls != null && conferenceCalls.size() > 0){
-                    return generateCallStatus(getJsonProperty(conferenceCalls.get(0), "id", String.class), CallStatus.CallStatusState.Connected);
+            String deviceDialString = retrieveDeviceDialString();
+            Pattern pattern = Pattern.compile("(\\d+):(\\d+):.+$", Pattern.CASE_INSENSITIVE);
+            Matcher matcher = pattern.matcher(callId);
+            if (!StringUtils.isNullOrEmpty(callId) && matcher.find()) {
+                Integer connectionId = Integer.parseInt(matcher.group(2));
+                try {
+                    JsonNode response = doGet(String.format(CONFERENCE, matcher.group(1)), JsonNode.class);
+                    if (response == null) {
+                        return generateCallStatus(callId, CallStatus.CallStatusState.Disconnected);
+                    }
+                    Boolean conferenceIsActive = getJsonProperty(response, "isActive", Boolean.class);
+                    if (conferenceIsActive != null) {
+                        // Generating callId by given conference response and target connectionId
+                        return generateCallStatus(generateCallId(response, connectionId, deviceDialString),
+                                conferenceIsActive ? CallStatus.CallStatusState.Connected : CallStatus.CallStatusState.Disconnected);
+                    }
+                } catch (CommandFailureException cfe) {
+                    if (cfe.getStatusCode() == 404) {
+                        return generateCallStatus(callId, CallStatus.CallStatusState.Disconnected);
+                    }

```

we should not swallow the exception if it is not 404. Is there a reason you are ignoring it if not 404?

---

yes, if there's no such call on the device - the call is disconnected. So essentially it's one of the valid scenarios. Should I add warn log here?

---

warn is fine